File: discover_ruby/github_most_popular.rb
require 'HTTPClient'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 39fed9cadb8579194e82b4bc4aefa34b8fab30b7'
}


clnt = HTTPClient.new
puts clnt.get_content('https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc', query = nil, extheaders = {})

File: discover_more_python/github_most_popular.py
import urllib2

request_headers = {
  'User-Agent': 'Holberton_School',
  'Authorization': 'token 6be002d3b7968b5ef3fbbb318e97afe71225afe5'
}  

# req = urllib2.Request('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc')
# response = urllib2.urlopen(req)
# the_page = response.read()
# data = {}

# response = urllib2.urlopen('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc')
req = urllib2.Request('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc',  None, request_headers)
response = urllib2.urlopen(req)
the_page = response.read()
print (req)